java/client/src/main/java/glide/api/models/commands/StreamTrimOptions.java

```diff
@@ -0,0 +1,155 @@
+/** Copyright GLIDE-for-Redis Project Contributors - SPDX Identifier: Apache-2.0 */
+package glide.api.models.commands;
+
+import glide.api.commands.StreamBaseCommands;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import lombok.NonNull;
+import lombok.RequiredArgsConstructor;
+
+/**
+ * Optional arguments for {@link StreamBaseCommands#xadd(String, Map, StreamAddOptions)} and {@link
+ * StreamBaseCommands#xtrim(String, TrimLimit)}
+ *
+ * @see <a href="https://redis.io/commands/xtrim/">redis.io</a>
+ */
+public final class StreamTrimOptions {
+    public static final String TRIM_MAXLEN_REDIS_API = "MAXLEN";
+    public static final String TRIM_MINID_REDIS_API = "MINID";
+    public static final String TRIM_EXACT_REDIS_API = "=";
+    public static final String TRIM_NOT_EXACT_REDIS_API = "~";
+    public static final String TRIM_LIMIT_REDIS_API = "LIMIT";
+
+    public abstract static class TrimLimit {
+
+        /**
+         * If <code>true</code>, the stream will be trimmed exactly. Equivalent to <code>=</code> in the
+         * Redis API. Otherwise, the stream will be trimmed in a near-exact manner, which is more
+         * efficient, equivalent to <code>~</code> in the Redis API.
+         */
+        protected boolean exact;
```

Comment:
Do we want this to be required? Redis defaults to exact trimming when unspecified, which doesn't seem to be an option here.

Reply:
Right.  It is optional.  Will fix.